src/routes/github/create-branch/github-branches-get.ts

```diff
@@ -17,8 +17,15 @@ export const GithubBranchesGet = async (req: Request, res: Response): Promise<vo
 
 	try {
 		const gitHubUserClient = await createUserClient(githubToken, jiraHost, req.log, gitHubAppConfig.gitHubAppId);
-		const branches = await gitHubUserClient.getReferences(owner, repo, 100);
-		res.send(branches);
+		const [ branches, repository ] = await Promise.all([
+			gitHubUserClient.getReferences(owner, repo),
+			gitHubUserClient.getRepository(owner, repo)
+		]);
+
+		res.send(branches.data.map(branch => ({
```

Comment:
We can determine the default as part of getReferences, I reckon it would be a bit cleaner.

Then we can drop the `getRepository` call maybe??

The getReferences graphQL is returning defaultBranchRef name, so we could just mutate the response in `getReferences` with default boolean like your map is doing.
```
 defaultBranchRef {
        name
      } 
```

If thats annoying to reference, we can always slap the default ref in each branch node with:
**In getBranchesNameQuery**
```

edges {
        node {
          name
          repository {
                      defaultBranchRef {
                        name
                      }
                    }
```

Reply:
Using `defaultBranchRef` in the Graphql throws 403 error for any repository that is not connected/synced to GH4J yet. That is why an added REST request is done.

Reply:
ahhh sorry, you are most correct. forgot about that!